### DISCUSSION TITLE
all values seem to be selected

### DISCUSSION THREAD

**Discussion by Bayi-Mostapha** [shadcn-ui/ui#2958] [General]
I am trying to make a select input for countries, I fetch them from an api and display them, it works fine for a regular select:
`<select {...register('country')}>`
`    {countries.map((country, index) => (`
`        <option key={index} value={country.cioc}>{country.name.common}</option>`
`    ))}`
`</select>`

![Screenshot 2024-03-09 000056](https://github.com/shadcn-ui/ui/assets/133959392/a0988b01-5735-46c6-9558-43fa531f8a1d)

but for the **shadcn select**, it does not work:

`<Select {...register('country')}>`
`    <SelectTrigger>`
`        <SelectValue placeholder="select a country" />`
`    </SelectTrigger>`
`    <SelectContent>`
`        {countries.map((country, index) => (`
`            <SelectItem key={index} value={country.cioc}>{country.name.common}</SelectItem>`
`        ))}`
`    </SelectContent>`
`</Select>`

<hr>

initially it seems like all values are selected, and it ruins my ui:
![Screenshot 2024-03-08 235822](https://github.com/shadcn-ui/ui/assets/133959392/388bcc2e-a8c1-43fa-a7d2-ae786c49d36e)

and even if i choose a value, the register does not seem to work (the string stays empty):
![Screenshot 2024-03-08 235854](https://github.com/shadcn-ui/ui/assets/133959392/00a5a852-f4d3-416a-83d0-f16e33f70ca6)

**Comment by KDust7** [comment]
For me, this error happens when more than one <SelectItem> has the same value. It may be a different issue, however when I was experiencing this type of behavior, this was the case.
